packages/component-meta/scripts/generateComponentMeta.ts

```diff
@@ -220,10 +236,43 @@ const generateComponentMeta = (): void => {
       requiredProps.push(invalidLinkUsageProp);
     }
 
-    let [, rawPropTypes] = /const [a-z][a-zA-z]+: (?:Omit<)?PropTypes<.+?> = ({[\s\S]+?});/.exec(source) || [];
-    rawPropTypes = rawPropTypes
-      ?.replace(/([a-zA-Z]+): (.+),/g, '$1: "$2",') // wrap values in quotes to make the object evaluable
-      .replace(/[^"](AllowedTypes\.(?:shape|oneOf).+\([{[][\s\S]+?[}\]]\)),/g, '`$1`,'); // wrap multiline shape object and oneOf array in backticks
+    let [, rawPropTypes] = /const [a-z][a-zA-Z]+: (?:Omit<)?PropTypes<.+?> = ({[\s\S]+?});/.exec(source) || [];
+
+    // handle shared props
+    const sharedPropsName = rawPropTypes?.match(/(?<=\.{3})(([a-zA-Z]+)(?=,))/g);
+    if (sharedPropsName) {
+      // TODO: currently the scenario for multiple shared props is not supported
+      if (sharedPropsName.length > 1) {
+        throw new Error('Currently the scenario for multiple shared props is not supported.');
+      }
+      const sharedPropsConstName = sharedPropsName[0];
+      const sharedPropsExportFilePath = getExportFilePath(source, sharedPropsConstName, tagName);
+
+      const sharedPropsExportFile = fs.readFileSync(`${sharedPropsExportFilePath}.ts`, 'utf8');
+      let [, sharedProps] = /const [a-z][a-zA-Z]+: .+? = ({[\s\S]+?});/.exec(sharedPropsExportFile) || [];
+
+      // add imports of shared properties
+      // TODO: currently the scenario for multiple shared props is not supported
+      // TODO: currently no other imports than from packages/components/src/utils/index.ts are supported
+      const allSharedConstants = sharedPropsExportFile.match(/([A-Z]{2,}(_[A-Z]+)*)/g).join(', ');
+      source = `import { ${allSharedConstants} } from '../../utils';\n${source}`;
+
+      // since it is more verbose to address shared proptypes using require(),
+      // the next steps are to ensure the extracted object matches the required type
+      const evaluableSharedProps = getEvaluablePropTypeString(sharedProps);
+      const requiredKeys = Object.keys(require(sharedPropsExportFilePath)[sharedPropsConstName]).sort();
+      const readFileKeys = Object.keys(eval(`(${evaluableSharedProps})`) as Record<string, string>).sort();
+      // TODO: currently the scenario for shared props imported from multiple files is not supported
+      if (JSON.stringify(requiredKeys) !== JSON.stringify(readFileKeys)) {
+        throw new Error('Currently the scenario for shared props imported from multiple files is not supported.');
+      }
+
+      sharedProps = sharedProps.replace(/([{}])/g, '');
+      rawPropTypes = rawPropTypes.replace(/\.{3}[a-zA-Z]+,/g, sharedProps);
```

Comment:
How can we be sure that we are actually replacing the right sharedProps? Shouldn't we use the previously extracted string?